### PR TITLE
Add Scheduler evictor for ConsumerGroup

### PR DESCRIPTION
**Pull request by pierDipi**
Part of #1537

## Proposed Changes

- Add Scheduler evictor for ConsumerGroup

### CODE REVIEW

File: control-plane/pkg/reconciler/consumergroup/evictor.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2022 The Knative Authors
```
Comment (matzew) [inline control-plane/pkg/reconciler/consumergroup/evictor.go:2]: was this updated? :smiley_cat:

File: control-plane/pkg/reconciler/consumergroup/evictor.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		Get(e.ctx, key.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get consumer group %s/%s: %w", key.Namespace, key.Name, err)
+	}
+
+	// Do not evict when scheduling is in-progress.
+	if cg.IsNotScheduled() {
```
Comment (matzew) [inline control-plane/pkg/reconciler/consumergroup/evictor.go:83]: is scheduling really in progress? or maybe not yet in progress too?

File: control-plane/pkg/reconciler/consumergroup/evictor.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to update consumer group %s/%s: %w", cg.GetNamespace(), cg.GetName(), err)
+	}
+
+	return nil
+}
+
+func (e *evictor) markPodUnschedulable(logger *zap.Logger, pod *corev1.Pod) error {
```
Comment (matzew) [inline control-plane/pkg/reconciler/consumergroup/evictor.go:102]: `disablePodScheduling` ? could we pass the `pod` as first arg?

File: control-plane/pkg/reconciler/consumergroup/evictor.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/evictor.go:83]: Right!

File: control-plane/pkg/reconciler/consumergroup/evictor.go (hunk 3 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/evictor.go:102]: Sure

File: control-plane/pkg/reconciler/consumergroup/evictor.go (hunk 2 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/evictor.go:83]: Done

File: control-plane/pkg/reconciler/consumergroup/evictor.go (hunk 3 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/evictor.go]: Done

File: control-plane/pkg/reconciler/consumergroup/evictor.go (hunk 1 above)
Comment (pierDipi) [reply control-plane/pkg/reconciler/consumergroup/evictor.go:2]: The boilerplate? no we didn't but the check is year-independent

**Review by aavarghese — APPROVED** [review]
Went through the code and all looks great!!

Would have been nice to have some code like `disablePodScheduling` in the common scheduler code for re-use but not critical for now.

/lgtm

**Pull request merged**